src/Target/LazyRoute.php

```diff
@@ -77,4 +77,33 @@ public function generate (UrlGeneratorInterface $urlGenerator) : string
     {
         return $urlGenerator->generate($this->route, $this->parameters, $this->referenceType);
     }
+
+
+    /**
+     * @return static
+     */
+    public function withParameters (array $parameters) : self
+    {
+        $modified = clone $this;
+        $modified->parameters = \array_replace($modified->parameters, $this->normalizeParameters($parameters));
+        return $modified;
+    }
+
+
+    /**
+     * Normalizes the parameters
+     */
+    private function normalizeParameters (array $parameters) : array
+    {
+        $normalized = [];
+
+        foreach ($parameters as $key => $value)
+        {
+            $normalized[$key] = \method_exists($value, "getId")
```

Comment:
```suggestion
            // automatically integrate with entities, so that you can just pass the entity and it will automatically use its id
            $normalized[$key] = \is_object($value) && \method_exists($value, "getId")
```